database/spanner_test.go

```diff
@@ -69,3 +74,19 @@ func Test_MigrateAndRun(t *testing.T) {
 	defer rows.Close()
 	require.NoError(t, rows.Err())
 }
+
+func TestSpannerUniqueViolation(t *testing.T) {
+	errMsg := "Failed to insert row with primary key ({pk#primary_key:\"282f6ffcd9ba5b029afbf2b739ee826e22d9df3b\"}) due to previously existing row"
+	// Test backwards-compatible parsing of spanner.Error (soon to be deprecated) from Spanner client
+	oldSpannerErr := spanner.ToSpannerError(status.New(codes.AlreadyExists, errMsg).Err())
+	if !database.SpannerUniqueViolation(oldSpannerErr) {
+		t.Error("should have matched unique violation")
+	}
+
+	// Test new apirerror.APIError response from Spanner client
+	newSpannerErr, parseErr := apierror.FromError(status.New(codes.AlreadyExists, errMsg).Err())
```

Comment:
nit: you can dedupe `status.New(codes.AlreadyExists, errMsg)`

Reply:
done